pkgs/backy/nautilus.nix

```diff
@@ -3,24 +3,39 @@
   poetry2nix ? pkgs.poetry2nix,
   fetchFromGitHub ? pkgs.fetchFromGitHub,
   lzo ? pkgs.lzo,
+  python310 ? pkgs.python310,
   ...
 }:
-let src = fetchFromGitHub {
-  owner = "flyingcircusio";
-  repo = "backy";
-  # TODO: this points to a branch head that only differs by the Nautilus-specific commits
-  # from what we use for Luminous, but not the latest `main` head as that currently
-  # fails to build in NixOS.
-  rev = "7914012a4e74dfeca597f9dcde237de5ee2f41b0";
-  sha256 = "sha256-EOWqQa//6UhvIIy3LGsFapmK2LMIBIQ90Om4XGOxS2g=";
-};
+let
+# src = ../path/to/checkout;
+  src = fetchFromGitHub {
+    owner = "flyingcircusio";
+    repo = "backy";
+    # TODO: this points to a branch head that only differs by the Nautilus-specific commits
+    # from what we use for Luminous, but not the latest `main` head as that currently
+    # fails to build in NixOS.
+    rev = "7914012a4e74dfeca597f9dcde237de5ee2f41b0";
+    sha256 = "sha256-EOWqQa//6UhvIIy3LGsFapmK2LMIBIQ90Om4XGOxS2g=";
+  };
+
 in poetry2nix.mkPoetryApplication {
     projectDir = src;
     src = src;
     doCheck = true;
+    python = python310;
+    extras = [];
     overrides = poetry2nix.overrides.withDefaults (self: super: {
       python-lzo = super.python-lzo.overrideAttrs (old: {
         buildInputs = [ lzo ];
       });
+      telnetlib3 = super.telnetlib3.overrideAttrs (old: {
+        buildInputs = [ super.setuptools ];
+      });
+      pytest-flake8 = super.pytest-flake8.overrideAttrs (old: {
+        buildInputs = [ super.setuptools ];
+      });
+      consulate = super.consulate.overrideAttrs (old: {
```

Comment:
For fc-qemu, we use a consulate built from a more recent commit than the last release, as upstream has not made a release for quite a while.
A manual test of backy on a dev host [indicates](https://github.com/flyingcircusio/backy/pull/28#issuecomment-1410777562) that now consulate 0.6 is too old for use in backy as well. Additionally, consulate currently has code that is deprecated in python-3.10, see https://github.com/osnyx/consulate/commit/f05c5b177e03c7e14e5f92c0026d5802c1348810 for a fix suggestion.